lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -499,7 +499,11 @@ def reconfig_redirect_opts!(res, opts)
     end
 
     # Don't forget any GET parameters
-    opts['query'] ||= location.query if location.query
+    if location.query
+      opts['query'] = location.query
+    else
+      opts['query'] = ''
+    end
```

Comment:
```suggestion
    opts['query'] = location.query
```

Reply:
Looks like we default to `''` in the bowels of rex, not a blocker 

https://github.com/rapid7/metasploit-framework/blob/c3cf56f06f5d56933b6226fdbf9ccec8ff8dcf82/lib/rex/proto/http/client_request.rb#L31